File: malipo-php/src/Resources/Transactions.php
<?php

namespace Malipo\Resources;

use Malipo\Malipo;

class Transactions
{
    private $client;

    public function __construct(Malipo $client)
    {
        $this->client = $client;
    }

    public function retrieve(string $id)
    {
        return $this->client->request('GET', '/transaction-status', ['id' => $id]);
    }
}

File: malipo-flutter/lib/src/resources/transactions.dart
import '../client.dart';
import '../models.dart';

class TransactionsResource {
  final Malipo client;

  TransactionsResource(this.client);

  Future<MalipoTransaction> retrieve(String id) async {
    final response = await client.request(
      'GET',
      '/transaction-status?id=$id',
    );

    return MalipoTransaction.fromJson(response);
  }
}
